pkg/commandutils/interfaces/fetcher_interface.go

```diff
@@ -0,0 +1,34 @@
+package interfaces
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+)
+
+//go:generate mockery -all -case=underscore
+
+// Fetcher Interface for exposing the fetch capabilities from the admin and also allow this to be injectable into other
+// modules. eg : create execution which requires to fetch launchplan details to construct the execution spec.
+type Fetcher interface {
```

Comment:
This is very counter intuitive I feel. We have to implement all methods of Admin in here, then why need it. We already have an interface for AdminService?

Reply:
This acts an Extension interface for getting data using AdminServiceClient for flytectl.

Advantages of using this pattern.
1] Simple interface and directly maps to use cases of the commands (eg : all version, latest version or given version)
    This can also be extended to be driven from the config . 
    eg : GetLPUsingCommandConfig which helps to translate the command level config and call the admin api's appropriately
2] Easier testing . The consuming command function become more testable with clear higher level interface instead of relying on admin service mocks. Also reduces testing code and makes sure unit test dont span multiple files as much as possible.
3] Dependency can be used by multiple commands for using the fetch capabilities and they need not redefine or have a dependency on another commands utility functions.